chaindataloader/binary.go

```diff
@@ -0,0 +1,227 @@
+package chaindataloader
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/neutrino/headerfs"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+/*
+
+================================================================================
```

Comment:
I think we can compress this format slightly by moving it as a normal comment to above the definition of `binReader`, so part as the normal `godoc` comment. This way, if we decide to make it public in the future, then that'll show up on anything that renders `godoc`.